mlflow/protos/service.proto

```diff
@@ -3069,3 +3109,77 @@ message Scorer {
   // The creation time of the scorer version (in milliseconds since epoch).
   optional int64 creation_time = 5;
 }
+
+// =============================================================================
+// Prompt Optimization Messages
+// =============================================================================
+
+// Request to start a prompt optimization job.
+message OptimizePrompts {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // ID of the training dataset.
+  optional string train_dataset_id = 1 [(validate_required) = true];
+  
+  // ID of the evaluation dataset.
+  optional string eval_dataset_id = 2 [(validate_required) = true];
+  
+  // URL of the prompt to optimize.
+  optional string prompt_url = 3 [(validate_required) = true];
+  
+  // List of scorer names to use for evaluation.
+  repeated string scorers = 4 [(validate_required) = true];
+
+    // Target LLM to use for optimization.
+  optional string target_llm = 5;
+
+  // Algorithm to use for optimization.
+  optional string algorithm = 6;
+
+  message Response {
+    // ID of the created optimization job.
+    optional string job_id = 1;
+  }
+}
+
+// Request to get the status and result of a prompt optimization job.
+message GetOptimizePromptsJob {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // ID of the optimization job.
+  optional string job_id = 1 [(validate_required) = true];
+
+  // Status of a prompt optimization job.
+  enum PromptOptimizationJobStatus {
+    PROMPT_OPTIMIZATION_JOB_STATUS_UNSPECIFIED = 0;
```

Comment:
q: do we need to care about the enum key conflict even if the enum is defined inside a message?

Reply:
yeah, that's my understanding. So I wonder if we should just use `UNSPECIFIED` for simplicity here